### PR TITLE
[go-gen] Add util.go generation for both auth and non-auth services

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds `util.go` generation to auth service
* Updates `util.go` generation for non-auth services
* Moves `config` struct from own file to inside `util.go` to follow Go-style locality
  * This diverges from spec-golang
* Updates tests

In the next PR I'm gonna move the folders around so they make more sense, i.e. `service/go` in the Scala project will become `server/go` with `auth`, `common` and `service` subdirectories.

You can ignore `.snippet` files while reviewing! Only 74 lines to review!

### CODE REVIEW

File: src/main/resources/go/genFiles/common/util/create_error_json.go.snippet [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// CreateErrorJSON returns a JSON string containing the key error associated with provided value
+func CreateErrorJSON(message string) string {
+	payload := map[string]string{"error": message}
+	json, err := json.Marshal(payload)
+	if err != nil {
+		return err.Error()
+	}
+	return string(json)
```
Comment (Lewiky) [inline src/main/resources/go/genFiles/common/util/create_error_json.go.snippet:8]: Why is this a string?

File: src/main/resources/go/genFiles/common/util/get_config.go.snippet [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// GetConfig returns a configuration object from decoding the given configuration file
+func GetConfig(filePath string) (*Config, error) {
+	config := Config{}
+	file, err := os.Open(filePath)
```
Comment (Lewiky) [inline src/main/resources/go/genFiles/common/util/get_config.go.snippet:4]: what does Open() do?

File: src/main/resources/go/genFiles/util/auth_struct.go.snippet [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Auth contains the unique identifier for a given auth
+type Auth struct {
+	ID uuid.UUID
```
Comment (Lewiky) [inline src/main/resources/go/genFiles/util/auth_struct.go.snippet:3]: OOH OOH EYE D

**Review by jaylees14 — APPROVED** [review]

File: src/main/resources/go/genFiles/util/ids_from_request.go.snippet [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	// Extract claims from JWT
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return nil, errors.New("JWT claims are invalid")
+	}
+
+	// Extract ID from JWT claims
+	id, ok := claims["id"]
+	if !ok {
+		return nil, errors.New("JWT does not contain an id")
```
Comment (Lewiky) [inline src/main/resources/go/genFiles/util/ids_from_request.go.snippet:35]: Wht if it does contain an id

File: src/main/resources/go/genFiles/util/ids_from_request.go.snippet [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	token, _, err := jwtParser.ParseUnverified(rawToken, jwt.MapClaims{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract claims from JWT
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return nil, errors.New("JWT claims are invalid")
+	}
```
Comment (Lewiky) [inline src/main/resources/go/genFiles/util/ids_from_request.go.snippet:30]: Is this close brace needed?

**Review by Lewiky — APPROVED** [review]

**Pull request merged**